packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -218,6 +218,104 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
+        relayRequestAncillaryData[keccak256(ancillaryData)] = depositHash;
+
+        // Sanity check that pool has enough balance to cover relay amount + proposer reward. Reward amount will be
+        // paid on settlement after the OptimisticOracle price request has passed the challenge period.
+        uint256 proposerBond = _getProposerBond(depositData.amount);
+
+        require(
+            l1Token.balanceOf(address(this)) >= depositData.amount + proposerBond &&
+                liquidReserves >= depositData.amount + proposerBond,
+            "Insufficient pool balance"
+        );
+
+        // Pull relay amount minus fees from caller and send to the deposit l1Recipient. The total fees paid is the sum
+        // of the LP fees, the relayer fees and the instant relay fee.
+        uint256 feesTotal =
+            _getAmountFromPct(
+                relayData.realizedLpFeePct + depositData.slowRelayFeePct + depositData.instantRelayFeePct,
+                depositData.amount
+            );
+
+        // Compute total proposal bond and pull from caller so that the OptimisticOracle can pull it from here.
+        l1Token.safeTransferFrom(
+            msg.sender,
+            address(this),
+            depositData.amount + proposerBond + store.computeFinalFee(address(l1Token)).rawValue - feesTotal
+        );
+
+        // Request a price for the relay identifier and propose "true" optimistically. This method will pull the
+        // (proposer reward + proposer bond + final fee) from the caller. We need to set a new price request timestamp
+        // instead of default setting to equal to the `depositTimestamp`, which is dependent on the L2 VM on which the
+        // DepositContract is deployed. Imagine if the timestamps on the L2 have an offset that are always "in the
+        // future" relative to L1 blocks, then the OptimisticOracle would always reject requests.
+        _requestAndProposeOraclePriceRelay(proposerBond, priceRequestTime, ancillaryData);
+
+        pendingReserves += depositData.amount; // Book off maximum liquidity used by this relay in the pending reserves.
+
+        bytes32 instantRelayHash = _getInstantRelayHash(depositHash, relayData);
+        require(
+            // Can only speed up a pending relay without an existing instant relay associated with it.
+            instantRelays[instantRelayHash] == address(0),
+            "Relay cannot be sped up"
+        );
+        instantRelays[instantRelayHash] = msg.sender;
+
+        // If the L1 token is WETH then: a) pull WETH from instant relayer b) unwrap WETH c) send ETH to recipient.
+        uint256 recipientAmount = depositData.amount - feesTotal;
+        if (isWethPool) {
+            WETH9Like(address(l1Token)).withdraw(recipientAmount);
+            depositData.l1Recipient.transfer(recipientAmount);
+        }
+        // Else, this is a normal ERC20 token. Send to recipient.
+        else l1Token.safeTransfer(depositData.l1Recipient, recipientAmount);
```

Comment:
this chunk of code is repeated. refactor to internal method?

Reply:
Will do!

Reply:
Actually, I think there are subtle differences that make this a little bit less tractable.